contracts/contracts/VanityURL.sol

```diff
@@ -0,0 +1,220 @@
+    /// @param _revenueAccount revenue account address
+    function setRevenueAccount(address _revenueAccount) public onlyOwner {
+        emit RevenueAccountChanged(revenueAccount, _revenueAccount);
+
+        revenueAccount = _revenueAccount;
+    }
+
+    /// @notice Set a new URL
+    /// @dev If the domain is expired, all the vanity URL info is erased
+    /// @dev If the domain ownership is transferred but not expired, all the vanity URL info is kept
+    /// @param _name domain name
+    /// @param _aliasName alias name for the URL
+    /// @param _url URL address to be redirected
+    /// @param _price Price to paid for the URL access
+    function setNewURL(
+        string calldata _name,
+        string calldata _aliasName,
+        string calldata _url,
+        uint256 _price
+    ) external payable nonReentrant whenNotPaused onlyDCOwner(_name) whenDomainNotExpired(_name) {
+        require(bytes(_aliasName).length <= 1024, "VanityURL: alias too long");
+        require(bytes(_url).length <= 1024, "VanityURL: url too long");
+
+        require(!checkURLValidity(_name, _aliasName), "VanityURL: url already exists");
+
+        uint256 price = urlUpdatePrice;
+        require(price <= msg.value, "VanityURL: insufficient payment");
+
+        // set a new URL
+        bytes32 tokenId = keccak256(bytes(_name));
+        vanityURLs[tokenId][_aliasName] = _url;
+        vanityURLPrices[tokenId][_aliasName] = _price;
+        vanityURLUpdatedAt[tokenId][_aliasName] = block.timestamp;
+
+        // returns the exceeded payment
+        uint256 excess = msg.value - price;
+        if (excess > 0) {
+            (bool success, ) = msg.sender.call{value: excess}("");
+            require(success, "cannot refund excess");
+        }
+
+        emit NewURLSet(msg.sender, _name, _aliasName, _url, _price);
+    }
+
+    /// @notice Delete the existing URL
+    /// @dev Deleting the URL is available regardless the domain expiration
+    /// @param _name domain name
+    /// @param _aliasName alias name for the URL to delete
+    function deleteURL(string calldata _name, string calldata _aliasName) external whenNotPaused onlyDCOwner(_name) {
+        require(checkURLValidity(_name, _aliasName), "VanityURL: url not exist");
+
+        bytes32 tokenId = keccak256(bytes(_name));
+        string memory url = vanityURLs[tokenId][_aliasName];
+
+        // delete the URL
+        vanityURLs[tokenId][_aliasName] = "";
+        vanityURLPrices[tokenId][_aliasName] = 0;
+        vanityURLUpdatedAt[tokenId][_aliasName] = block.timestamp;
+
+        emit URLDeleted(msg.sender, _name, _aliasName, url);
+    }
+
+    /// @notice Update the existing URL
+    /// @dev Updating the URL is not available if the domain is expired
+    /// @param _name domain name
+    /// @param _aliasName alias name for the URL
+    /// @param _url URL address to be redirected
+    /// @param _price Price to paid for the URL access
+    function updateURL(
```

Comment:
It seems this can be merged with setNewURL

Reply:
Hmm... it can be. However, updateURL function should be called when the url doesn't exist and it doesn't require payment. We might merge with setNewURL function by adding one flag parameter. But if you're fine, I'd like to have both of them separately.